src/main.cpp

```diff
@@ -2080,6 +2081,47 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+Consensus::VersionBitsConditionChecker checkCSV(Consensus::DEPLOYMENT_CSV);
```

Comment:
The reason I didn't put them in versionbits is exactly because the lock is
in main. Post c++11 we can move locks to libconsensus, and the versionbits
cache can have its own lock.